tiledb/sm/consolidator/group_meta_consolidator.cc

```diff
@@ -76,58 +76,60 @@ Status GroupMetaConsolidator::consolidate(
   // Open group for reading
   auto group_uri = URI(group_name);
   Group group_for_reads(group_uri, storage_manager_);
-  RETURN_NOT_OK(group_for_reads.open(
-      QueryType::READ, config_.timestamp_start_, config_.timestamp_end_));
+  group_for_reads.open(
+      QueryType::READ, config_.timestamp_start_, config_.timestamp_end_);
 
   // Open group for writing
   Group group_for_writes(group_uri, storage_manager_);
-  RETURN_NOT_OK_ELSE(
-      group_for_writes.open(QueryType::WRITE),
-      throw_if_not_ok(group_for_reads.close()));
+  try {
+    group_for_writes.open(QueryType::WRITE);
+  } catch (const std::exception&) {
```

Comment:
Better to avoid a selector and use `catch(...)`, especially since it's rethrowing.

Reply:
Ah, so that's how C++ does it. C# has just `catch` which didn't work here. Thanks!